### ISSUE TITLE
Handle programmatic need for resolution better

### ISSUE THREAD

**Issue by oxinabox** [oxinabox/DataDeps.jl#8]
make this not be required if you have a variable that holds part of the data dep name;

```
function word_embeddings(langcode, path=nothing)
    path = if path_ isa Void
        DataDeps.resolve(DataDeps.registry["FastText $langcode"]) # This line is bad. Fix this.
    else
        path_
    end
end
```

Maybe via String Interpolatation. (see implementation in StringInterning.jl)

**Comment by oxinabox** [comment]
I think `resolve(name::AbstractString)=resolve(registry(name))` should be added

**Comment by oxinabox** [comment]
I am pretty sure `@datadeps_str "FastText $langcode"` works. but it is ugly

**Comment by oxinabox** [comment]
I'm going to say this was closed in #41 and if I am still annoyed reopen it later
